src/core/crypto/ecdsa.hpp

```diff
@@ -202,6 +238,11 @@ class P256
         {
             return otPlatCryptoEcdsaSign(this, &aHash, &aSignature);
         }
+
+#if !OPENTHREAD_CONFIG_PLATFORM_KEY_REFERENCES_ENABLE
+    private:
+        uint8_t mBytes[OT_CRYPTO_ECDSA_MAX_DER_SIZE];
```

Comment:
We should not add new member variable in sub-classes when they are mirror of a C-style public/platform `struct`.

Here we have `class KeyPair : public otPlatCryptoEcdsaKeyPair` mirroring `otPlatCryptoEcdsaKeyPair` to a C++ style class. We want to be able to treat any `otPlatCryptoEcdsaKeyPair` as ` KeyPair.

Reply:
Good point. Chanign back the implementation in next push.